pkg/token/options.go

```diff
@@ -236,3 +250,29 @@ func getCacheFileName(o *Options) string {
 	}
 	return filepath.Join(o.TokenCacheDir, fmt.Sprintf(cacheFileNameFormat, o.Environment, o.ServerID, o.ClientID, o.TenantID))
 }
+
+// Parses the pop token claims. Pop token claims are passed in as a comma-separated string
+// in the format "key1=val1,key2=val2".
+func parsePopClaims(popClaims string) (map[string]string, error) {
+	if strings.TrimSpace(popClaims) == "" {
+		return nil, fmt.Errorf("error parsing PoP token claims: no claims provided")
+	}
+	claimsArray := strings.Split(popClaims, ",")
+	claimsMap := make(map[string]string)
+	for _, claim := range claimsArray {
+		claimPair := strings.Split(claim, "=")
+		if len(claimPair) < 2 {
+			return nil, fmt.Errorf("error parsing PoP token claims. Ensure the claims are formatted as `key=value` with no extra whitespace")
+		}
+		key := strings.TrimSpace(claimPair[0])
+		val := strings.TrimSpace(claimPair[1])
+		if key == "" || val == "" {
+			return nil, fmt.Errorf("error parsing PoP token claims. Ensure the claims are formatted as `key=value` with no extra whitespace")
+		}
+		claimsMap[key] = val
+	}
+	if claimsMap["u"] == "" {
+		return nil, fmt.Errorf("required u-claim not provided for PoP token flow. Please provide the ARM ID of the connected cluster in the format `u=<ARM_ID>`")
```

Comment:
same remark, we should just mention `cluster` not `connected cluster`

Reply:
Updated to just say 'cluster'